pkg/model/types_test.go

```diff
@@ -92,6 +96,24 @@ func TestNewEndpointFromInstance(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name: "missing ServiceType",
+			inst: &types.HttpInstanceSummary{
+				InstanceId: &instId,
+				Attributes: map[string]string{
+					EndpointIpv4Attr:      ip,
+					EndpointPortAttr:      "80",
+					EndpointProtocolAttr:  "TCP",
+					EndpointPortNameAttr:  "http",
+					ServicePortNameAttr:   "http",
+					ServiceProtocolAttr:   "TCP",
+					ServicePortAttr:       "65535",
+					ServiceTargetPortAttr: "80",
+					"custom-attr":         "custom-val",
+				},
+			},
+			wantErr: true,
```

Comment:
Are we sure we want an error if the type is not in Cloud Map?

Reply:
I thought since `ServiceImport` relies on the `ServiceType` attribute, it would be necessary to have?

Reply:
The cluster/clusterset ID PR has similar functionality, so I think we should at least discuss if this is desirable. I'll leave a comment there as well.
It means we would have to remove all previously exported endpoints (and services) from Cloud Map before upgrading the controller. If two clusters are exporting services to each other, you could not upgrade either controller without removing both from the cluster set.
An alternative would be to fall back on current functionality if not present (assume clusterip type).

Reply:
Actually, it looks like we are okay here. If the endpoint cannot be extracted from a Cloud Map instance we simply skip it.
https://github.com/aws/aws-cloud-map-mcs-controller-for-k8s/blob/main/pkg/cloudmap/client.go#L238

Reply:
So in this case, we're good to keep the test as is?